### ISSUE TITLE
separate out some example/pset-only dependencies into poetry extras

### ISSUE THREAD

**Issue by RussTedrake** [RussTedrake/manipulation#390]
as opposed to dependencies that are required for the main manipulation _library_.

[Poetry now supports extras](https://python-poetry.org/docs/pyproject/#extras), which should be compatible with pip.

Essentially, if users want to include the core manipulation library tools in another repo, they shouldn't have to pull down all of pytorch, etc, in order to get e.g. `MakeHardwareStation`.

**Comment by nepfaff** [comment]
Agreed. There have been multiple comments that `manipulation` is too big to be used as a library.
